src/zc/queue/_queue.py

```diff
@@ -235,32 +235,23 @@ def __iter__(self):
                 yield i
 
     def __getitem__(self, index):
+        # not efficient, but quick and easy
         if isinstance(index, slice):
-            start, stop, stride = index.indices(len(self))
-            res = []
-            stride_ct = 1
-            for ix, v in enumerate(self):
-                if ix >= stop:
-                    break
-                if ix < start:
-                    continue
-                stride_ct -= 1
-                if stride_ct == 0:
-                    res.append(v)
-                    stride_ct = stride
-            return res
+            # We do this the easy way instead of attempting to handle
+            # negative strides ourself
+            return list(self)[index]
```

Comment:
I don't see it [explicitly in the documentation](https://docs.python.org/2.7/library/itertools.html#itertools.islice) but `islice` doesn't handle negative steps:
```
>>> import itertools
>>> itertools.islice((), None, None, -1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Step for islice() must be a positive integer or None.
```